scipy/stats/tests/test_continuous.py

```diff
@@ -550,50 +562,65 @@ def check_nans_and_edges(dist, fname, arg, res):
     outside_arg_plus = (outside_arg == 1) & valid_parameters
     endpoint_arg_minus = (endpoint_arg == -1) & valid_parameters
     endpoint_arg_plus = (endpoint_arg == 1) & valid_parameters
+
+    is_discrete = isinstance(dist, DiscreteDistribution)
```

Comment:
It would be nice to factor things differently, but settled on making the smallest changes to account for the discrete case, since refactoring is something that we should discuss together and shouldn't be required to get your PR landed.

There are skips and changes in reference values below contingent on whether the distribution is discrete.